client/src/components/Auth/AdminView/index.tsx

```diff
@@ -3,9 +3,9 @@ import React, { FC } from 'react';
 import { useAuth } from '../../../contexts';
 
 const AdminView: FC = ({ children }) => {
-  const { isAdmin } = useAuth();
+  const { isAdmin, user } = useAuth();
 
-  return <>{isAdmin && children}</>;
+  return <>{isAdmin && user.hasRoleApproved && children}</>;
```

Comment:
Would having the `isAdmin` variable be false if user.hasRoleApproved be false make more sense here? That way, when you check if `isAdmin` is true you've got a guarantee that the user is an admin an that they have admin permissions.

Reply:
You make a good point, Andrew! I think having that confirmation with isAdmin and isStaff would be nice to have.